test/m1_facilitator/handlers/DeclaredWithdrawIntentsHandler/handle.test.ts

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as web3Utils from 'web3-utils';
+
+import BigNumber from 'bignumber.js';
+import Repositories
+  from '../../../../src/m1_facilitator/repositories/Repositories';
+import DeclaredWithdrawIntentsHandler
+  from '../../../../src/m1_facilitator/handlers/DeclaredWithdrawIntentsHandler';
+import MessageRepository
+  from '../../../../src/m1_facilitator/repositories/MessageRepository';
+import WithdrawIntentRepository
+  from '../../../../src/m1_facilitator/repositories/WithdrawIntentRepository';
+import assert from '../../../test_utils/assert';
+import Message, {
+  MessageStatus,
+  MessageType,
+} from '../../../../src/m1_facilitator/models/Message';
+
+describe('DeclaredWithdrawIntentsHandler::handle', () => {
+  let handler: DeclaredWithdrawIntentsHandler;
+  let messageRepository: MessageRepository;
+  let withdrawIntentRepository: WithdrawIntentRepository;
+
+  const withdrawIntentEntityRecord = {
+    messageHash: web3Utils.sha3('1'),
+    contractAddress: '0x0000000000000000000000000000000000000001',
+    utilityTokenAddress: '0x0000000000000000000000000000000000000002',
+    amount: '2',
```

Comment:
```suggestion
    amount: new BigNumber('2'),
```

Reply:
Input from the subgraph will be string not BN.